handler/health.go

```diff
@@ -20,6 +22,30 @@ func HandleHealth() http.HandlerFunc {
 			Version: version,
 			OK:      true,
 		}
-		WriteJSON(w, response, http.StatusOK)
+		status := http.StatusOK
+
+		if !checkInternetConnectivity() {
+			response = api.HealthResponse{
+				Version: version,
+				OK:      false,
+			}
+			status = http.StatusInternalServerError
+		}
+
+		WriteJSON(w, response, status)
+	}
+}
+
+func checkInternetConnectivity() bool {
+	dialer := net.Dialer{
```

Comment:
I am wary about this change.
What if VMs are coming up but the network is slow. In that case, connectivity check will always fail here and previously passing health checks will now start to fail.

Reply:
we can increase the timeout to 2 seconds.